pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go

```diff
@@ -330,13 +330,13 @@ func SelectCandidate(candidates []Candidate) Candidate {
 	victimsMap := candidatesToVictimsMap(candidates)
 	candidateNode := pickOneNodeForPreemption(victimsMap)
 
-	// Same as candidatesToVictimsMap, this logic is not applicable for out-of-tree
-	// preemption plugins that exercise different candidates on the same nominated node.
-	for _, candidate := range candidates {
-		if candidateNode == candidate.Name() {
-			return candidate
+	if victims, _ := victimsMap[candidateNode]; victims != nil {
```

Comment:
Thanks for the catch. Please keep the comments.

Reply:
```suggestion
	if victims := victimsMap[candidateNode]; victims != nil {
```

Reply:
Thanks, updated